phpBB/includes/acp/acp_storage.php

```diff
@@ -182,11 +403,59 @@ public function overview($id, $mode)
 			];
 		}
 
+		$this->template->assign_vars([
+			'STORAGES'			=> $this->storage_collection,
+			'STORAGE_STATS'		=> $storage_stats,
+			'PROVIDERS' 		=> $this->provider_collection,
+
+			'U_ACTION'			=> $this->u_action . '&amp;hash=' . generate_link_hash('acp_storage'),
+		]);
+	}
+
+	protected function display_progress_bar()
+	{
+		adm_page_header($this->lang->lang('STORAGE_UPDATE_IN_PROGRESS'));
+		$this->template->set_filenames(array(
+			'body'	=> 'progress_bar.html')
+		);
 		$this->template->assign_vars(array(
-			'STORAGES' => $this->storage_collection,
-			'STORAGE_STATS' => $storage_stats,
-			'PROVIDERS' => $this->provider_collection,
-		));
+			'L_PROGRESS'			=> $this->lang->lang('STORAGE_UPDATE_IN_PROGRESS'),
+			'L_PROGRESS_EXPLAIN'	=> $this->lang->lang('STORAGE_UPDATE_IN_PROGRESS_EXPLAIN'))
+		);
+		adm_page_footer();
+	}
+
+	function close_popup_js()
+	{
+		return "<script type=\"text/javascript\">\n" .
```

Comment:
I'm really not a fan of having HTML inside a PHP file. There should be better ways of doing this.

Reply:
Me neither, but is the same code than acp_search

Reply:
acp_search has changed in the mean time